### ISSUE TITLE
upstream?

### ISSUE THREAD

**Issue by dimpase** [gap-system/homebrew-gap#6]
What does prevent this from being upstreamed?

**Comment by olexandr-konovalov** [comment]
Could you please clarify what does "upstreamed" mean?

**Comment by dimpase** [comment]
making GAP formula available on the main Homebrew repo, naturally.

**Comment by dimpase** [comment]
e.g. Pari/GP is there:
https://formulae.brew.sh/formula/pari#default

**Comment by olexandr-konovalov** [comment]
Ok (I had two other interpretations of "upstreamed" in mind, none of them matching yours :)

It was there before, in homebrew-science, but then after its deprecation we were recommended to maintain own tap.

**Comment by dimpase** [comment]
I don't know who recommended this, why not just make a PR to put it into the main repo? How is GAP more niche than Pari/GP, so that it should not be in the main repo ?

**Comment by olexandr-konovalov** [comment]
Can't find where that was recommended: IIRC it was suggested as an alternative in https://github.com/Homebrew/homebrew-science/pull/6500, but can't be sure, and the link is not working any more. That was already after I've attempted https://github.com/Homebrew/homebrew-core/pull/21928, which was not merged, so whoever will try this again, please look at the feedback at that PR first.

**Comment by dimpase** [comment]
as soon as GAP gets `make install` - which that PR promised in  4.10, it'd
become trivial...

**Comment by fingolfin** [comment]
Yeah, as soon as someone puts in the work to make it possible and resolve all the icky little bits and details, and also cleans this formula up to match Homebrew's quality requirement, it will indeed be trivial...

**Comment by fingolfin** [comment]
Oh and on that PR I did not *promise* to add `make install` support for 4.10 -- I stated it as a plan; like so many plans, it didn't succeed. Although we are a lot closer *shrug*

**Comment by dimpase** [comment]
Oops, I grew up with 5-year Plans blasting from every mass media orifice, which of course were _more, much more than promises_, so I probably am irreparably brain-damaged and take plans as promises, sorry.

**Comment by olexandr-konovalov** [comment]
For the record, two other conjectures of how should I interpret the verb "upstreamed" were:
a) update this repository to GAP 4.11.1.
b) make this a standard GAP distribution updated and announced together with the release (so part of the main GAP repository?)

(a) is in TODOs, and I 'm happy to go without (b) in either forms.

**Comment by fingolfin** [comment]
To quote the Homebrew documentation:

> We frown on authors submitting their own work unless it is very popular.

Take that as you will... *shrug*

**Comment by dimpase** [comment]
I think it's popular enough to be considered for inclusion (and it'd reduce traffic on the support and other help lists). The usual question, of course, is what packages to bundle, and how to arrange for the rest of the packages to be installable.

**Comment by fingolfin** [comment]
Here is the situation: if someone wants to try to get an official `gap` homebrew package up and running, *and maintain it*, they are of course free to do so. They can take whatever they want from this repo, or start from scratch, or whatever.

But this repo is unmaintained, and I am sorely tempted to delete it, but apparently it is useful for some people despite its deficiencies, so I am not.

**Comment by dimpase** [comment]
Singular is in [core Homebrew](https://formulae.brew.sh/formula/singular) - it seems without involvement of its authors.

Macaulay2 got its [Homebrew tab](https://github.com/Macaulay2/homebrew-tap) a while ago:
- it's about 15 Homebrew formulae for some of their dependencies (they have a lot of dependencies, some relatively obscure), and documentation.

The only design question for GAP formula is how to package the GAP packages. These  depending on extra external libraries/programs (e.g. semigroups) ought to be in separate formulae, but the bulk (GAP-only code) can probably be bumped into one formula.
